### PR TITLE
Feature/4278-Change Password Dialog

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes # [BLUI-4278](https://jira-prod.tcc.etn.com/browse/BLUI-4278).

#### Changes proposed in this Pull Request:

- Added ChangePasswordDialog component with translations

#### Screenshots / Screen Recording (if applicable)

Screenshot 2023-06-22 at 7 57 23 PM

#### To Test:

- yarn start:example2
- Click on `Change Password Dialog` button

#### Any specific feedback you are looking for?

-

### CODE REVIEW

**Comment by surajeaton** [comment]
@daileytj @JeffGreiner-eaton @ektaghag-eaton Ready for review.

**Comment by manojleaton** [comment]
Code looks good to me

**Comment by manojleaton** [comment]
Code looks good and working fine.

**Review by surajeaton — COMMENTED** [review]
@manojleaton Please resolve the conflicts.

**Review by ektaghag-eaton — COMMENTED** [review]
When I click on next button, nothing happened.
Screenshot 2023-06-29 at 11 09 21 AM

**Comment by ektaghag-eaton** [comment]
Console error in changePasswordDialogBase
Screenshot 2023-06-29 at 11 13 13 AM

File: login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        render(
+            <AuthContextProvider {...authContextProps}>
+                <BrowserRouter>
+                    <ChangePasswordDialog {...props} />
+                </BrowserRouter>
+            </AuthContextProvider>
+        );
+
+    it('renders without crashing', () => {
+        renderer();
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.test.tsx:33]: We are just rendering here, we need assertion to test that it renders without crashing, something like:
expect(something).toMatch(something)

File: login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    });
+
+    it('Current password input test', () => {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.test.tsx:36]: tese cases statements should be like example:
it('should display the default values')

File: login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.test.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            currentPasswordLabel: 'Current Password',
+        });
+
+        const currentPasswordInput = getByLabelText('Current Password');
+        expect(currentPasswordInput).toHaveValue('');
+        fireEvent.change(currentPasswordInput, { target: { value: 'Abc@2023' } });
+        expect(currentPasswordInput).toHaveValue('Abc@2023');
+    });
+
+    it('New password input fields test', () => {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.test.tsx:47]: Same comment as above

File: login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    return (
+        <ChangePasswordDialogBase
+            loading={isLoading}
+            dialogTitle={dialogTitle}
+            dialogDescription={dialogDescription}
+            currentPasswordLabel={currentPasswordLabel}
+            previousLabel={previousLabel}
+            nextLabel={nextLabel}
+            currentPasswordChange={currentPasswordChange}
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.tsx:95]: `currentPasswordChange` function is only being used once, so no need to define a function, you can directly setState here:
`currentPasswordChange={(currentPwd) => setCurrentInput(currentPwd)}`

File: login-workflow/src/new-architecture/components/ChangePasswordDialog/types.ts [hunk 5]
```diff
@@ -1,14 +1,19 @@
-import { DialogProps } from '@mui/material';
+import { SxProps } from '@mui/material';
+import { BasicDialogProps } from '../Dialog';
 import { SetPasswordProps } from '../SetPassword';
 
-// we should define a reusable type for the password input props (this should be defined with the reusable password component and reused by CreatePassword, ResetPassword, and ChangePassword).
-export type ChangePasswordDialogProps = DialogProps & { PasswordProps: SetPasswordProps } & {
+export type ChangePasswordDialogProps = { passwordProps?: SetPasswordProps } & {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/types.ts:5]: We need `DialogProps` for `ChangePasswordDialogProps`, as the root comp is Dialog, it should accept all the props of dialog.

Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/types.ts:5]: `passwordProps` should be `PasswordProps`. When we are defining a prop which will be available for any other child component, we are defining a camel case. Please `PasswordProps` in other components.
Same comment for errorDialogProps

File: login-workflow/src/new-architecture/components/ChangePasswordDialog/types.ts [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
     currentPasswordLabel?: string;
     previousLabel?: string;
     nextLabel?: string;
-    currentPasswordChange: (currentPassword: string) => void;
-    enableButton: boolean | (() => boolean);
-    onSubmit: () => void;
+    currentPasswordChange?: (currentPassword: string) => void;
+    enableButton?: boolean | (() => boolean);
+    onSubmit?: () => void;
+    onPrevious?: () => void;
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/types.ts:16]: last 3 props will come from `DialogProps` if you don't remove that.

File: login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    };
+
+    useEffect(() => {
+        setPasswordInput(areValidMatchingPasswords() ? passwordInput : '');
+    }, [setPasswordInput, passwordInput, confirmInput, areValidMatchingPasswords]);
+
+    const checkPasswords =
+        passwordInput !== '' && confirmInput !== '' && currentInput !== '' && passwordInput === confirmInput;
+
+    const changePasswordSubmit = (): void => {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.tsx:47]: This should be async and await on action. Please refer https://github.com/etn-ccis/blui-react-workflows/blob/ddce712207359d5b404dd1fdc7954f44ce680baa/login-workflow/src/new-architecture/screens/EulaScreen/EulaScreen.tsx#L29

**Review by surajeaton — APPROVED** [review]

**Pull request merged**